### PR TITLE
docs: added examples for datazone_domain

### PR DESCRIPTION
**Pull request by quixoticmonk**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request
* The resources and data sources in this provider are generated from the CloudFormation schema, so they can only support the actions that the underlying schema supports. For this reason submitted bugs should be limited to defects in the generation and runtime code of the provider. Customizing behavior of the resource, or noting a gap in behavior are not valid bugs and should be submitted as enhancements to AWS via the CloudFormation Open Coverage Roadmap.

### CODE REVIEW

File: examples/resources/awscc_datazone_domain/datazone_domain.tf [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        }
+      }
+    ]
+  })
+  managed_policy_arns = ["arn:aws:iam::aws:policy/service-role/AmazonDataZoneDomainExecutionRolePolicy"]
+}
+
+variable "account_id" {
+  type        = string
+  description = "AWS account id"
```
Comment (wellsiau-aws) [inline examples/resources/awscc_datazone_domain/datazone_domain.tf:43]:
```suggestion
  description = "Source AWS account id"
```

nit pick, I believe this should be called the source account id

File: examples/resources/awscc_datazone_domain/datazone_domain.tf [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            "aws:TagKeys" : "datazone*"
+          }
+        }
+      }
+    ]
+  })
+  managed_policy_arns = ["arn:aws:iam::aws:policy/service-role/AmazonDataZoneDomainExecutionRolePolicy"]
+}
+
+variable "account_id" {
```
Comment (wellsiau-aws) [inline examples/resources/awscc_datazone_domain/datazone_domain.tf:41]:
```suggestion
variable "source_account_id" {
```

File: examples/resources/awscc_datazone_domain/datazone_domain.tf [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        "Principal" : {
+          "Service" : "datazone.amazonaws.com"
+        },
+        "Action" : [
+          "sts:AssumeRole",
+          "sts:TagSession"
+        ],
+        "Condition" : {
+          "StringEquals" : {
+            "aws:SourceAccount" : var.account_id
```
Comment (wellsiau-aws) [inline examples/resources/awscc_datazone_domain/datazone_domain.tf:29]:
```suggestion
            "aws:SourceAccount" : var.source_account_id
```

**Review by wellsiau-aws — CHANGES_REQUESTED** [review]
Please take a look at the suggested change and correct my assumption as needed. Thanks!

**Review by wellsiau-aws — APPROVED** [review]
LGTM, thanks !

**Comment by marcosentino** [comment]
@quixoticmonk Thank you for the contribution!

**Pull request merged**